WordPress/gutenberg
# Mobile: Improve npm clean scripts for react-native-editor

## Description
Updating the clean scripts to delete the proper cache files.

It seems like these cache files have moved (possibly due to one of the React Native upgrades) since these scripts were last updated. I am basing these updates on the files that I am seeing on my local machine. Let me know if you think that these might be different for different people.

I did notice that there was also a `kotlin-compiler-in-wordpressreactnativebridge-17156846827643056472.alive` in the `$TMPDIR`. I am not adding anything to remove that at this time since I don't know what it is used for, but it might be something we want to consider in the future.

## Testing Instructions
1. Run `npm run bundle` from gutenberg-mobile
2. Observe that there is at least one `haste-map-metro-*` file and a `metro-cache/` file in "$TMPDIR"
3. Run `npm run clean` from gutenberg-mobile
4. Observe that the files from step 2 no longer exist, and that there are no other recently added or updated files relating to react-native, metro, or haste in `$TMPDIR/` (apart from the earlier mentioned `kotlin-compiler-in...` file.


## Types of changes
Infrastructure improvement

## Checklist:
